src/braket/circuits/duration_gate.py

```diff
@@ -0,0 +1,203 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+import math
+from collections.abc import Sequence
+from enum import Enum
+from functools import singledispatch
+from typing import Optional, Union
+
+from braket.circuits.free_parameter_expression import FreeParameterExpression
+from braket.circuits.gate import Gate
+from braket.circuits.parameterizable import Parameterizable
+
+
+class SiTimeUnit(Enum):
+    """Possible Si unit time types"""
+
+    s = "s"
+    ms = "ms"
+    us = "us"
+    ns = "ns"
+
+    def __str__(self):
+        return self.value
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+
+class DurationGate(Gate, Parameterizable):
+    """Class `DurationGate` represents a quantum gate that operates on N qubits and a duration."""
```

Comment:
Just a note, there is quite a bit of duplication between this `DurationGate` class and the existing `AngledGate` class. Did you consider refactoring to make a common base class for instructions with a single `float` parameter? I guess this would also ideally extend to the `DoubleAngledGate` and `TripleAngledGate` classes - so maybe outside the scope of this PR and could be tracked by a separate issue.